src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,128 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ */
+package com.mcmoddev.mmdbot.modules.logging.misc;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.mcmoddev.mmdbot.MMDBot;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageUpdateEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.Color;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class ScamDetector extends ListenerAdapter {
+
+    public static final Gson GSON = new GsonBuilder().disableHtmlEscaping().create();
+    public static final String SCAM_LINKS_DATA_URL = "https://phish.sinking.yachts/v2/all";
+
+    public static final List<String> SCAM_LINKS = Collections.synchronizedList(new ArrayList<>());
+
+    static {
+        new Thread(ScamDetector::setupScamLinks, "Scam link collector").start();
+    }
+
+    @Override
+    public void onGuildMessageReceived(@NotNull final GuildMessageReceivedEvent event) {
+        final var msg = event.getMessage();
+        if (containsScam(msg)) {
+            final var member = msg.getMember();
+            final var msgChannel = msg.getTextChannel();
+            final var guild = msg.getGuild();
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            msg.delete().queue($ -> {
+                final var embed = new EmbedBuilder()
+                    .setTitle("Scam link detected!")
+                    .setDescription(String.format("User %s sent a scam link in %s! Their message was deleted, and they were muted!", member.getAsMention(),
+                        msgChannel.getAsMention()))
+                    .setColor(Color.RED)
+                    .setTimestamp(Instant.now())
+                    .setFooter("User ID: " + member.getIdLong())
+                    .setThumbnail(member.getEffectiveAvatarUrl());
+                getLoggingChannel(guild).sendMessageEmbeds(embed.build()).queue();
```

Comment:
Could we get the message text quoted in a message in the #deleted-messages channel, like how the requests deletion system does?